datafusion/src/sql/planner.rs

```diff
@@ -79,6 +79,22 @@ pub struct SqlToRel<'a, S: ContextProvider> {
     schema_provider: &'a S,
 }
 
+fn plan_compound(mut identifiers: Vec<String>) -> Expr {
+    if &identifiers[0][0..1] == "@" {
+        Expr::ScalarVariable(identifiers)
+    } else if identifiers.len() == 2 {
```

Comment:
worth considering a follow up PR to handle an edge case where user tries to access nested field with unqualified column `column.field`. for compound identifiers, we should probably check the first identify to see if it's a valid relation name.